src/app/components/popup-host/popup-host.component.ts

```diff
@@ -0,0 +1,35 @@
+import {Component, Input, TemplateRef, ViewChild, ViewContainerRef} from '@angular/core';
+
+@Component({
+    selector: 'app-popup-host',
+    templateUrl: './popup-host.component.html',
+    styleUrls: ['./popup-host.component.css'],
+})
+export class PopupHostComponent {
+    @ViewChild('viewPort', {read: ViewContainerRef, static: true})
+    private readonly viewPort: ViewContainerRef | undefined;
+
+    @Input() set popupTemplate(templateRef: TemplateRef<unknown> | null) {
+        if (templateRef) {
+            this.insertPopupTemplate(templateRef);
+        } else {
+            this.onClose();
+        }
```

Comment:
Чуть чуть попродаю best practics)

Не рекомендую к использованию конструкцию `if/else`, особенно в комбинации нескольких `else` (`if / else if / else`). Вместо этого рекомендую использовать ранний `return`, он более гибки позволяет делать промежуточны вычисления, и, соответсвенно, писать более чистый код)
```suggestion
        if (templateRef) {
            this.insertPopupTemplate(templateRef);
            
            return;
        }
        
        this.onClose();
```

Reply:
ну по идее несколько if/else вообще лучше заменить на switch, но согласна, исправила

Reply:
> ну по идее несколько if/else вообще лучше заменить на switch

Да, хорошая мысль!